.github/workflows/main.yaml

```diff
@@ -0,0 +1,22 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+      - name: Test
+        run: mvn --batch-mode test
+
+# TODO's
+#  - setup integration tests
+#     - these require an openAI (and hugging face, etc) token
```

Comment:
any idea how this can be done? some kind of secrets in github?

Reply:
Yes, github will let you expose secrets to your CI steps and whatever needs them in your tests while hiding them from the public.

(We should then make sure they don't leak in the logs in any way)